### PR TITLE
chore: Do not sass-compile external apps

### LINKED ISSUE #38092: [Bug]: Front-end build touches *all* apps scss/css
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Running `npm run build` touches app scss and creates unexpected css files

I raised this earlier in https://github.com/nextcloud/server/pull/35622#issuecomment-1339442060 already and a discussion spawned at https://github.com/nextcloud/standards/issues/9 but until a decision has been made and docs still recommend people to clone into `apps` I think it's fair to say the server front-end should leave unrelated apps untouched.

### Steps to reproduce

1. Clone server
2. Clone calendar into apps
3. Compile the server front-end

### Expected behavior

No changes in the calendar repo

### Installation method

None

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
N/a
```

### List of activated Apps

```shell
N/a
```

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

cc @skjnldsv @AndyScherzinger

### PR DESCRIPTION
**Pull request by ChristophWurst**
* Close https://github.com/nextcloud/server/issues/38092

## Summary

Only compile apps in this repo. Ignore ignored directories.

Discovered while rebuilding PRs that don't build on CI. I had to reset all external apps' `css/` directory because of stray sass compilation.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by ChristophWurst** [comment]
Forgot the /css suffix

**Comment by ChristophWurst** [comment]
so

**Review by Antreesy — APPROVED** [review]
It works for me locally: compiler ignore all the `apps/directory`, which are not included in `.gitignore`.

But `Psalm static code analysis` is still red, so it's up to you to decide

File: package.json [hunk 1]
```diff
@@ -17,8 +17,8 @@
     "test": "jest",
     "test:watch": "jest --watch",
     "test:jsunit": "karma start tests/karma.config.js --single-run",
-    "sass": "sass --style compressed --load-path core/css core/css/ apps/*/css",
-    "sass:watch": "sass --watch --load-path core/css core/css/ apps/*/css",
+    "sass": "sass --style compressed --load-path core/css core/css/ $(for cssdir in $(find apps -mindepth 2 -maxdepth 2 -name \"css\"); do if ! $(git check-ignore -q $cssdir); then echo -n \"$cssdir \"; fi; done)",
+    "sass:watch": "sass --watch --load-path core/css core/css/ $(for cssdir in $(find apps -mindepth 2 -maxdepth 2 -name \"css\"); do if ! $(git check-ignore -q $cssdir); then echo -n \"$cssdir \"; fi; done)",
```
[the remark this answers is not in the document]
Comment (ChristophWurst) [reply package.json:21]: > This way people have to add their additional apps to .gitignore though, right? Can we do something like [this](https://stackoverflow.com/questions/2405305/how-to-tell-if-a-file-is-git-tracked-by-shell-exit-code) instead?

Additional apps *are* ignored: https://github.com/nextcloud/server/blob/d80277c340eb737537a582112130f96e9361d7dd/.gitignore#L14-L43

[the remark this answers is not in the document]
Comment (marcelklehr) [reply package.json:21]: Ah, didn't know that. makes sense then

**Review by marcelklehr — APPROVED** [review]

**Pull request merged**